Average Risk Screening and Determine Next Due Date

mickohanlon23 opened the pull request:
This PR contains CQL logic and accompanying test cases for the Average Risk Screening and Determine Next Due Date flows. I also included AverageRisk.fsh with Ruleset templates for PlanDefintion actions, which can be built out further once we develop the PlanDefinition structure in the Decision to Screen flow.

ptran265 committed:
avg risk tests and cql first pass

ptran265 committed:
attempt at tiered approach parameters - throws errors

mickohanlon23 committed:
Temporarily skip failing AvgRisk Tests

mickohanlon23 committed:
Add ELM for new AvgRisk CQL

mickohanlon23 committed:
Add DataElements test stubs for DetermineNextDueDate path

mickohanlon23 committed:
Add CQL stubs for DetermineNextDueDate path

mickohanlon23 committed:
Add test stubs for flow-DetermineNextDueDate, DetermineNextDueDateActions, DetermineNextDueDateEvents

mickohanlon23 committed:
Data element CQL stubs for DetermineNextDueDate

mickohanlon23 committed:
Set up framework and add relevant codes for DetermineNextDueDateActions.cql

mickohanlon23 committed:
Create CQL stubs for DetermineNextDueDate.cql

mickohanlon23 committed:
Add DetermineNextDueDate PlanDefinition, and stub out PD actions

mickohanlon23 committed:
Try parameterizing TieredApproach

mickohanlon23 committed:
Remove tieredApproach event test fixture, since this will now live as a parameter. Add parameter to tiered approach test

mickohanlon23 committed:
Merge branch 'main' into determine-next-due-date

mickohanlon23 committed:
Add de-MostRecentTestIs___ fixtures to data_elements.yml

mickohanlon23 committed:
Change DataElements tests to include default patient

mickohanlon23 committed:
Update flow-DetermineNextDueDate tests based upon Feb '25 updates to DetermineNextDueDate path in L2

mickohanlon23 committed:
Remove 'Tiered Approach' tests and CQL Actions from Average Risk Path

mickohanlon23 committed:
Bring in latest CRCSMCommonFunction.cql library

mickohanlon23 committed:
Update Testing Modality terminology in DataElements.cql, and incorporate new way of querying for Testing Modality DiagnosticReports and Observations within DataElements.cql

mickohanlon23 committed:
Add tests for FOBT testing modality, and remove tess for gFOBT and FIT, which were combined into FOBT

mickohanlon23 committed:
Add CQL Expressions for MostRecentTestIs____

mickohanlon23 committed:
Add tests for DataElements/de-MostRecentTestIs____

mickohanlon23 committed:
Add tests for DataElements/de-stoolDNAFIT, so that all testing modality data elements have tests

mickohanlon23 committed:
Remove outdated test/DataElements test stubs, specifically for de-mostRecent____

mickohanlon23 committed:
Remove outdated 'de-MostRecentTestIs____' CQL expressions from DataElements.cql

mickohanlon23 committed:
Fix improperly formatted de-MostRecentTestIsDNAFIT test fixture alias in data_elements.yml

mickohanlon23 committed:
Build out DetermineNextDueDateEvents.cql and tests

mickohanlon23 committed:
Build function for determining next due date

mickohanlon23 committed:
Build out top level flow for DetermineNextDueDate.cql

mickohanlon23 committed:
Complete Average Risk Actions

mickohanlon23 committed:
Add tests for flow-DetermineNextDueDate

mickohanlon23 committed:
Remove repetitive test stubs for DetermineNextDueDateActions

mickohanlon23 committed:
Add tests for flow-USPSTFAvgRisk

mickohanlon23 committed:
Create template for AverageRisk PlanDefinition actions

ptran265 commented on input/cql/DetermineNextDueDateActions.cql:
```diff
@@ -0,0 +1,60 @@
+define MostRecentColorectalScreeningDate:
+  DataElements.MostRecentColorectalScreeningDate
+
+define function UpdateDueDate(screeningInteval System.Quantity):
```
typo - screeningInteval -> screeningInterval

mickohanlon23 replied:
Good catch, fixed in https://github.com/CancerScreeningCDS/crcsm-cds/pull/21/commits/c26a86c7b85fd8472cf33bc52e78b02a774ee83f.

ptran265 commented on input/cql/DetermineNextDueDateEvents.cql:
```diff
@@ -0,0 +1,23 @@
+
+context Patient
+
+define "eve-lastTestColonoscopyAvg":
```
I'm assuming this event name came from the L2? If so, it's fine but was just wondering why this name has 'Avg' and the other modalities don't

mickohanlon23 replied:
Yeah that comes directly from the events table in the L2: https://gitlab.mitre.org/cancer-cds/colorectal-cancer/cdc-colorectal-cancer-screening-cds-l2/-/blob/main/input/data/events.csv?ref_type=heads&plain=1#L49. I wasn't sure why it was there either. In https://github.com/CancerScreeningCDS/crcsm-cds/commit/c26a86c7b85fd8472cf33bc52e78b02a774ee83f, I went ahead and did a find and replace to remove the "Avg" part.

ptran265 commented on test/flow-DetermineNextDueDate/eve-lastTestFlexSig_no_end-1.yml:
```diff
@@ -0,0 +1,12 @@
+  - *patientage45to76range
+results:
+  flow-DetermineNextDueDate:
+    null
```
should the output of this be null or should it be recommending 'due now'?

ptran265 commented:
Looks good, just need to update the sushi config with the new flows (under groups) and the input/pagecontent/4_examples.md page so the new content will render on the IG. Also we should either make the reason codes more specific (as discussed earlier) or add a TODO for later.

mickohanlon23 committed:
Address review comments

mickohanlon23 committed:
Add USPSTFAvgRisk group to sushi-config.yaml. Generate FHIR Library resources from CQL

mickohanlon23 committed:
Update examples page links with latest flow-USPSTFAvgRisk test cases.

mickohanlon23 committed:
Add TODOs in USPSTFAvgRiskActions.cql

mickohanlon23 committed:
Delete extraneous nonsensical test cases

mickohanlon23 commented:
> Looks good, just need to update the sushi config with the new flows (under groups) and the input/pagecontent/4_examples.md page so the new content will render on the IG. Also we should either make the reason codes more specific (as discussed earlier) or add a TODO for later.

I updated sushi config, added my tests to the examples page, added TODOs for the reasonCode and the timingAge, and removed extra nonsensical tests. @ptran265 This is ready for re-review.
